include/boost/decimal/detail/cmath/log.hpp

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 Matt Borland
+// Copyright 2023 Christopher Kormanyos
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+
+#ifndef BOOST_DECIMAL_DETAIL_CMATH_LOG_HPP
+#define BOOST_DECIMAL_DETAIL_CMATH_LOG_HPP
+
+#include <cmath>
+#include <type_traits>
+
+#include <boost/decimal/fwd.hpp> // NOLINT(llvm-include-order)
+#include <boost/decimal/detail/type_traits.hpp>
+
+namespace boost { namespace decimal {
+
+template<typename T>
+constexpr auto log(T x) noexcept -> std::enable_if_t<detail::is_decimal_floating_point_v<T>, T> // NOLINT(misc-no-recursion)
+{
+    // TODO(ckormanyos) Handle arguments infinity and NaN.
+
+    constexpr auto zero = T { 0 };
+    constexpr auto one  = T { 1 };
+
+    auto result = T { };
+
+    if (isinf(x) || isnan(x))
+    {
+        result = x;
+    }
+    else if (x < one)
+    {
+        // Handle reflection.
+        result = ((x > zero) ? -one / log(-x) : -std::numeric_limits<T>::infinity());
+    }
+    else if(x > one)
+    {
+        constexpr auto two  = T { 2 };
+
+        // The algorithm for logarithm is based on Chapter 5, pages 35-36
+        // of Cody and Waite, Software Manual for the Elementary Functions,
+        // Prentice Hall, 1980.
+
+        auto exp2val = int { };
```

Comment:
`int exp2val {}` seems like the cleaner option in the cases where you don't initialize the type with something besides default. That is the way I have been going.

Reply:
I was wondering which way was best/faster/better. I'll try to be consistent to your suggestion. If I miss a few, we can re-edit them.